gcc/rust/checks/errors/rust-unsafe-checker.cc

```diff
@@ -784,7 +784,23 @@ UnsafeChecker::visit (Trait &trait)
 void
 UnsafeChecker::visit (ImplBlock &impl)
 {
-  // FIXME: Handle unsafe impls
+  bool safe = !impl.is_unsafe ();
+  // Check for unsafe-only attributes on generics and lifetimes
+  if (safe)
+    for (auto &parm : impl.get_generic_params ())
+      {
+	for (auto o_attr : (*parm).get_outer_attrs ())
```

Comment:
```suggestion
	for (auto o_attr : parm->get_outer_attrs ())
```

any reason for the deref + method call instead of this?

Reply:
Fixed